devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy

```diff
@@ -0,0 +1,346 @@
+	def result = []
+
+	if (cmd.notificationType == 0x08) {
+		def value = cmd.event== 0x03? "on" : "off"
+		endpoint ? result = changeSwitch(endpoint, value) : []
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.crc16encapv1.Crc16Encap cmd) {
+	log.info "zwaveEvent Crc16Encap called: cmd '${cmd}'"
+	def versions = commandClassVersions
+	def version = versions[cmd.commandClass as Integer]
+	def ccObj = version ? zwave.commandClass(cmd.commandClass, version) : zwave.commandClass(cmd.commandClass)
+	def encapsulatedCommand = ccObj?.command(cmd.command)?.parse(cmd.data)
+	if (encapsulatedCommand) {
+		zwaveEvent(encapsulatedCommand)
+	}
+	[:]
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensormultilevelv5.SensorMultilevelReport cmd) {
+	log.info "zwaveEvent SensorMultilevelReport called, ${cmd}"
+	def map = [:]
+	def result = []
+	switch (cmd.sensorType) {
+		case 1:
+			map.name = "temperature"
+			def cmdScale = cmd.scale == 1 ? "F" : "C"
+			map.value = convertTemperatureIfNeeded(cmd.scaledSensorValue, cmdScale, cmd.precision)
+			map.unit = getTemperatureScale()
+			break		
+		case 5:
+			map.name = "humidity"
+			map.value = cmd.scaledSensorValue.toInteger()
+			map.unit = "%"
+			break
+		default:
+			map.descriptionText = cmd.toString()
+	}
+	log.debug "SensorMultilevelReport, ${map}, ${map.name}, ${map.value}, ${map.unit}"
+	result << createEvent(map)
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd, endpoint = null) {
+	log.info "zwaveEvent ***** Unhandled Command called, cmd '${cmd}', endpoint '${endpoint}' *****"
+	[descriptionText: "Unhandled $device.displayName: $cmd", isStateChange: true]
+}
+
+
+/**
+ * PING is used by Device-Watch in attempt to reach the Device
+ * */
+def ping(endpoint = null) {
+	log.info "ping called"
+	refresh(endpoint)
+}
+
+def refresh(endpoint = null) {
+	log.info "refresh called: endpint '${endpoint}' "
+	if(endpoint) {
+		secureEncap(zwave.basicV1.basicGet(), endpoint)
+	} else {
+		def commands = []
+		commands << zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 1)
+		commands << zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 5)
+		sendCommands(commands,1000)
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv2.ConfigurationReport cmd){
+	log.info "zwaveEvent ConfigurationReport called: ${cmd}"
+	def result = []
+	switch (cmd.parameterNumber) {
+		case 1:
+			state.tempOffset = cmd.scaledConfigurationValue
+			break
+		case 2:
+			state.humidityOffset = cmd.scaledConfigurationValue
+			break
+	}
+	result <<  createEvent(name: "configReport", value: "${state.tempOffset}, ${state.humidityOffset}%")
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelEndPointReport cmd, endpoint = null) {
+	log.info "zwaveEvent MultiChannelEndPointReport called: cmd '${cmd}'"
+	if(!childDevices) {
+		def type = getNumberOfChildFromModel()
```

Comment:
nit: `type` is not a well-named variable here

Reply:
This part will update from "type" to "numberOfChild".